### PR TITLE
(feat) Advanced Workflow State Management

### PR DESCRIPTION
**Pull request by ZacButko**
This PR introduces a workflow state management system managed by the FormWorkflowReducer which triggers actions in the esm-form-entry-app AMPATH forms and listens for the response of posted data.

It is now possible to trigger form submission by clicking "next patient" or "review & save".

Also a review screen was added.

Please see video below

https://user-images.githubusercontent.com/5445264/180996414-cd5290ea-8b04-4bfc-a0e3-e2a5386847d2.mov

### CODE REVIEW

File: src/form-entry-workflow/styles.scss [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 
 .breadcrumbsContainer > div > div > nav {
-  background-color: white;
-  padding: 0.5rem;
-  border-bottom: 1px solid $openmrs-background-grey;
+  background-color: $ui-02;
+  padding: 0.75rem 1rem;
```
Comment (vasharma05) [inline src/form-entry-workflow/styles.scss:8]: $spacing-04 and $spacing-05

File: src/form-entry-workflow/styles.scss [hunk 2]
```diff
@@ -28,16 +28,23 @@
   display: flex;
   text-align: center;
   margin-top: 1rem;
+  column-gap: 1rem;
```
Comment (vasharma05) [inline src/form-entry-workflow/styles.scss:31]: $spacing-05

File: src/form-review-card/FormReviewCard.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            <div className={styles.dataField}>
+              <ul>
+                {encounter.obs.map((obs, index) => (
+                  <li key={index}>{obs.display}</li>
+                ))}
+              </ul>
+            </div>
+          )}
+          <Button kind="primary" onClick={() => editEncounter(patientUuid)}>
+            Go To Form
```
Comment (vasharma05) [inline src/form-review-card/FormReviewCard.tsx:39]: Translation please

File: src/form-review-card/styles.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.formReviewCard :global(.bx--accordion) :global(.bx--accordion__item) {
+  border: none;
+}
+
+.formReviewCard :global(.bx--accordion__title) {
+  display: flex;
+  align-items: baseline;
+  column-gap: 1rem;
```
Comment (vasharma05) [inline src/form-review-card/styles.scss:18]: $spacing-05

File: src/form-review-card/styles.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.formReviewCard :global(.bx--accordion__content) {
+  padding: 0.5rem;
```
Comment (vasharma05) [inline src/form-review-card/styles.scss:22]: $spacing-03

**Comment by vasharma05** [comment]
Some changes I think can be improved.

Also, please have a look at the missed translation above.
Everything else looks all good!

**Review by vasharma05 — APPROVED** [review]

File: src/form-review-card/styles.scss (hunk 5 above)
Comment (ZacButko) [reply src/form-review-card/styles.scss:22]: done

File: src/form-review-card/styles.scss (hunk 4 above)
Comment (ZacButko) [reply src/form-review-card/styles.scss:18]: done

File: src/form-entry-workflow/styles.scss (hunk 2 above)
Comment (ZacButko) [reply src/form-entry-workflow/styles.scss:31]: done

File: src/form-entry-workflow/styles.scss (hunk 1 above)
Comment (ZacButko) [reply src/form-entry-workflow/styles.scss:8]: done

**Comment by ZacButko** [comment]
Good eye, thanks @vasharma05 !

**Pull request merged**